sphinx/highlighting.py

```diff
@@ -73,14 +73,18 @@ def __init__(self, dest: str = 'html', stylename: str = 'sphinx',
             self.formatter = self.latex_formatter
             self.formatter_args['commandprefix'] = 'PYG'
 
-    def get_style(self, stylename: str) -> Style:
+    def get_style(self, stylename: Union[str, Style]) -> Style:
         if stylename is None or stylename == 'sphinx':
             return SphinxStyle
         elif stylename == 'none':
             return NoneStyle
         elif '.' in stylename:
```

Comment:
I guess a style object get a failure with this if-block. Is this PR well tested at local?
As I already commented, a testcase is needed to merge this.